cpp/src/slice2swift/SwiftUtil.cpp

```diff
@@ -615,8 +616,104 @@ SwiftGenerator::writeServantDocSummary(IceInternal::Output& out, const Interface
 void
 SwiftGenerator::validateMetadata(const UnitPtr& u)
 {
-    MetadataVisitor visitor;
-    u->visit(&visitor);
+    map<string, MetadataInfo> knownMetadata;
+
+    // "swift:attribute"
+    MetadataInfo attributeInfo = {
+        .validOn = {typeid(ClassDecl), typeid(Struct), typeid(Slice::Exception), typeid(Enum)},
+        .acceptedArgumentKind = MetadataArgumentKind::RequiredTextArgument,
+        .mustBeUnique = false,
+    };
+    knownMetadata.emplace("swift:attribute", attributeInfo);
+
+    // "swift:class-resolver-prefix"
+    MetadataInfo classResolverPrefixInfo = {
+        .validOn = {typeid(Unit)},
+        .acceptedArgumentKind = MetadataArgumentKind::SingleArgument,
+    };
+    knownMetadata.emplace("swift:class-resolver-prefix", classResolverPrefixInfo);
+
+    // "swift:inherits"
```

Comment:
We should drop swift:inherits.

Reply:
After this is merged I'll have a follow up PR to drop `swift:inherits` and loosen the `swift:module` check.
I think that should be all the issues around Swift metadata.